packages/babel-cli/src/babel/file.js

```diff
@@ -16,7 +16,10 @@ export default function(commander, filenames, opts) {
 
   const buildResult = function() {
     const map = new sourceMap.SourceMapGenerator({
-      file: path.basename(commander.outFile || "") || "stdout",
+      file:
+        commander.sourceMapTarget ||
```

Comment:
We also had weird cases like this, where if the option was in babel-core, then every file has their own target, making it unclear what the `file` field should be when using babel-cli to concatenate a bunch of files together. By implementing it in CLI itself, we know that the option is meant for the final output file.

This is the only place the CLI option is actually used now, which I think makes sense.